wasm/tests/test.c

```diff
@@ -51,8 +60,84 @@ void test_opa_malloc()
     }
 }
 
+void test_opa_malloc_min_size()
+{
+    reset_heap();
+
+    // Ensure that allocations less than the minimum size
+    // are creating blocks large enough to be re-used by
+    // the minimum size.
+    void *too_small = opa_malloc(4);
+    test("allocated min block", too_small != NULL);
+
+    void *barrier = opa_malloc(0);
```

Comment:
It wasn't immediately obvious to me why `barrier` is needed. Will the free blocks go to zero without `barrier`?

Reply:
right, the block would be removed from the free list without this.